### PR TITLE
CDAP-5578 Added a getEntityName method to entityId

### PR DESCRIPTION
**Pull request by riwazp7**
CDAP-5578 "Add getEntityName() to EntityId Class" https://issues.cask.co/browse/CDAP-5578

This method makes getting the name of an entity without doing <switch case to find what subclass it belongs to and then casting to that class> possible. This will remove a hack in Tracker. 

Passing build: http://builds.cask.co/browse/CDAP-DUT4750-1

### CODE REVIEW

**Comment by russorat** [comment]
LGTM, but we need to figure out why the checks are failing.

File: cdap-proto/src/main/java/co/cask/cdap/proto/id/ApplicationId.java [hunk 1]
```diff
@@ -46,6 +46,10 @@ public String getApplication() {
     return application;
   }
 
+  public String getEntityName() {
```
Comment (bdmogal) [inline cdap-proto/src/main/java/co/cask/cdap/proto/id/ApplicationId.java:50]: please add `@Override` annotation to these methods.

**Comment by riwazp7** [comment]
I'll try running a build.

**Comment by riwazp7** [comment]
Travis fails for some reason but bamboo passes. Probably some unrelated flaky test. Will merge.

**Pull request merged**